### ISSUE TITLE
Sorting and grouping tabs via AI

### ISSUE THREAD

**Issue by MStankiewiczOfficial** [mbnuqw/sidebery#1731]
### Description

A very useful feature would be the sorting and grouping of browser tabs using so-called artificial intelligence, such as OpenAI GPT-4 Mni or Google Gemini. So that there would be no cost to the extension developer, users would use their API keys to use this function.

**Comment by JohnVKRDam** [comment]
I think that this is entirely beyond the scope of this extension, to be honest.

**Comment by ric2b** [comment]
This is now going to be a native Firefox feature, not sure if any work is needed on this extension to work with it: https://support.mozilla.org/en-US/kb/how-use-ai-enhanced-tab-groups

**Comment by the-nelsonator** [comment]
Native tab groups support is tracked here: https://github.com/mbnuqw/sidebery/issues/1927. Whether this would come for free depends on implementation

**Comment by megamorphg** [comment]
This would be insanely useful for us tab hoarders. I also have one window I barely use with tabs I need to reorganize after the tabs got all messed up. Would be great if we can select a bunch of tabs and have it re-organize.
Also needs a "system prompt" to have it point to a template that tells how we typically like organizing our tabs.

It would be useful if we can point it to an LLM via API Keys or use the on-device LLM for tab reorganization inside of Sidebery.

**Comment by JohnVKRDam** [comment]
Nah, 3559 tabs and counting, and I have ZERO need for AI in my workflow. I actually have no interest at all in any AI anywhere. I am not going to allow it to infect my devices. AI integration is a deal breaker for me.

<img width="477" height="729" alt="Image" src="https://github.com/user-attachments/assets/9fce0f41-316d-47a8-936a-7970242cfdfc" />

**Comment by ric2b** [comment]
@JohnVKRDam You can just not use the feature, that's not a reason for it not to exist.

**Comment by JohnVKRDam** [comment]
It's not an existing feature in SIDEBERY, though. So, in my opinion it has no reason to exist. At all. And valuable developer time should not be wasted on it.

I also really don't care if you agree or not.
